app/models/course/registration.rb

```diff
@@ -0,0 +1,61 @@
+class Course::Registration < ActiveRecord::Base
+  stampable
+
+  scope :staff, -> { where(role: [roles[:teaching_assistant], roles[:manager], roles[:owner]]) }
+
+  enum role: CourseUser.roles
+
+  belongs_to :course
+  belongs_to :user
+
+  def staff?
+    teaching_assistant? || manager? || owner?
+  end
+
+  # Approve the registration record by creating a CourseUser of the user to the course with the
+  # requested role.
+  def approve
+    if course.course_users.create(user_id: user_id, role: role)
+      true
+    else
+      course.course_user.errors.each do |attr, message|
+        errors[attr] = message
+      end
+      false
+    end
+  end
+
+  # Process the registration records of given IDs and then delete them.
+  #
+  # @param [Course] the course that the registration records belong to.
+  # @param [Array<Int>] the IDs of registration records
+  # @options approve: If specified a true equivalent value, the Registration#approve method will
+  #                   be called. Default to false.
+  #          process_all: if specified a true equivalent value, all registration to the course
+  #                       will be processed. Default to false.
+  #          role: If specified a value in the Registration.roles enum, only registrations of that
+  #                role will be processed. Other IDs are ignored. Default to nil.
+  # @return [Hash] If successful, the hash contains an array of processed IDs with the key -
+  #                'processed_ids'. Otherwise it contains an ActiveModel::Errors with the key -
+  #                'errors'.
+  def self.process(course, registration_ids, options = {})
+    registrations = registrations_to_be_processed(course, registration_ids, options)
+    processed_ids = []
+    registrations.each do |registration|
+      return registration.errors if options[:approve] && !registration.approve
+      processed_ids << registration.id
+      registration.destroy!
+    end
+    { processed_ids: processed_ids }
+  end
+
+  def self.registrations_to_be_processed(course, registration_ids, options = {})
+    conditions = {}
+    conditions[:id] = registration_ids unless options[:process_all]
+    options[:role] = %w(teaching_assistant manager owner) if options[:role].to_s == 'staff'
```

Comment:
Hound complains about constant arrays and suggests %w. Maybe the setting is wrong?



> On 26 Mar 2015, at 8:30 am, Joel Low <notifications@github.com> wrote:
> 
> In app/models/course/registration.rb:
> 
> > +  #                'errors'.
> > +  def self.process(course, registration_ids, options = {})
> > +    registrations = registrations_to_be_processed(course, registration_ids, options)
> > +    processed_ids = []
> > +    registrations.each do |registration|
> > +      return registration.errors if options[:approve] && !registration.approve
> > +      processed_ids << registration.id
> > +      registration.destroy!
> > +    end
> > +    { processed_ids: processed_ids }
> > +  end
> > +
> > +  def self.registrations_to_be_processed(course, registration_ids, options = {})
> > +    conditions = {}
> > +    conditions[:id] = registration_ids unless options[:process_all]
> > +    options[:role] = %w(teaching_assistant manager owner) if options[:role].to_s == 'staff'
> I prefer constant arrays.
> 
> \
> Reply to this email directly or view it on GitHub.
>